packages/beacon-node/src/network/processor/gossipHandlers.ts

```diff
@@ -263,8 +263,9 @@ function getDefaultHandlers(modules: ValidatorFnsModules, options: GossipHandler
               const forkTypes = config.getForkTypes(slot);
               const rootHex = toHexString(forkTypes.BeaconBlock.hashTreeRoot(signedBlock.message));
 
+              logger.debug("Blobs are unavailable within cutoff time", {slot, root: rootHex, peer: peerIdStr});
               events.emit(NetworkEvent.unknownBlock, {rootHex, peer: peerIdStr});
-              break;
+              return;
```

Comment:
Returning here means we won't track this in metrics as a process block error and also not prune `seenGossipBlockInput`
https://github.com/ChainSafe/lodestar/blob/c52c7bd6614c622f77913e85ec435bf0f50e2558/packages/beacon-node/src/network/processor/gossipHandlers.ts#L281-L283

Maybe someone can clarify this, based on [seenGossipBlockInput.ts#L31-L42](https://github.com/ChainSafe/lodestar/blob/c52c7bd6614c622f77913e85ec435bf0f50e2558/packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts#L31-L42), I am not sure if it correct to even prune the cache?

Reply:
why don't we want to track it as an error in metric?
Also cache needs to be pruned in case there is some sort of DOS attack

Reply:
> why don't we want to track it as an error in metric? 
We throw an error but it used to trigger unknown block sync upstream
https://github.com/ChainSafe/lodestar/blob/83d7d701728ac70ad01cd80074b8537632ac0a23/packages/beacon-node/src/chain/blocks/verifyBlocksDataAvailability.ts#L121-L122
But that's based on my limited understanding, will revert that change to track an error

> Also cache needs to be pruned in case there is some sort of DOS attack

My thought process was we might prune a block input which was valid but just didn't receive blobs for it yet, but you are probably right that this might be unsafe to not prune always

https://github.com/ChainSafe/lodestar/blob/83d7d701728ac70ad01cd80074b8537632ac0a23/packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts#L71